File: spark/src/main/java/com/jason/spark/core/Spark01RDDOperatorTransform.java
package com.jason.spark.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

/**
 * @Description:
 * @author: 贾森
 * @date: 2024年10月24日 15:41
 */
public class Spark01RDDOperatorTransform {
    public static void main(String[] args) {
        SparkConf sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark01RDDOperatorTransform");
        JavaSparkContext sc = new JavaSparkContext(sparkConf);
        JavaRDD<Integer> parallelize = sc.parallelize(Arrays.asList(1, 2, 3, 4, 5));
        JavaRDD<Integer> mappedRDD = parallelize.map(num -> num * 2);
        mappedRDD.collect().forEach(System.out::println);
        sc.stop();
    }
}

File: spark/src/main/scala/com/jason/spark/core/rdd/Spark01_RDD_Operator_Transform.scala
package com.jason.spark.core

import org.apache.spark.{SparkConf, SparkContext}

object Spark01_RDD_Operator_Transform {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setMaster("local[*]").setAppName("create RDD from files")
    val sc = new SparkContext(sparkConf)
    val rdd = sc.makeRDD(List(1, 2, 3, 4, 5, 6))
    // val rdd2 = rdd.map(num => num * 2 )
    val rdd2 = rdd.map(_ * 2 )
    rdd2.collect().foreach(println)
    sc.stop()
  }

}
